user_management/api/serializers.py

```diff
@@ -8,18 +8,24 @@
 User = get_user_model()
 
 
-class ValidateEmailMixin(object):
-    def validate_email(self, attrs, source):
-        email = attrs.get(source).lower()
+class UniqueEmailValidator(validators.UniqueValidator):
+    def filter_queryset(self, value, queryset):
+        """Check lower-cased email is unique."""
+        return super(UniqueEmailValidator, self).filter_queryset(
+            value.lower(),
+            queryset,
+        )
 
-        try:
-            User.objects.get_by_natural_key(email)
```

Comment:
By removing `get_by_natural_key` project using `django-user-management` and modifying the email before being stored in the database would need to subclass the new `validator` to bring back `get_by_natural_key`.